materialize-snowflake/snowflake_test.go

```diff
@@ -11,66 +11,72 @@ import (
 	"github.com/estuary/flow/go/protocols/catalog"
 	pf "github.com/estuary/flow/go/protocols/flow"
 	pm "github.com/estuary/flow/go/protocols/materialize"
-	sqlDriver "github.com/estuary/flow/go/protocols/materialize/sql"
+	sql "github.com/estuary/connectors/materialize-sql"
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
 )
 
 func TestQueryGeneration(t *testing.T) {
 	var spec *pf.MaterializationSpec
 	require.NoError(t, testsupport.CatalogExtract(t, "testdata/flow.yaml",
-		func(db *sql.DB) error {
+		func(db *stdsql.DB) error {
 			var err error
 			spec, err = catalog.LoadMaterialization(db, "test/sqlite")
 			return err
 		}))
 
-	var gen = SQLGenerator()
-	var table = sqlDriver.TableForMaterialization("test_table", "", gen.IdentifierRenderer, spec.Bindings[0])
+	var shape1 = sql.BuildTableShape(spec, 0, tableConfig{
+		Table:  "test_table",
+		Delta:  false,
+	})
+	table, err := sql.ResolveTable(shape1, snowflakeDialect)
+	require.NoError(t, err)
 
 	var loadUUID = uuid.UUID{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}
 	var storeUUID = uuid.UUID{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1}
 
-	var keyJoinSQL, copyIntoSQL, mergeIntoSQL = BuildSQL(123, table, spec.Bindings[0].FieldSelection, loadUUID, storeUUID)
+	loadQuery, err := RenderTableWithRandomUUIDTemplate(table, loadUUID, tplLoadQuery)
+	require.NoError(t, err)
+	copyInto, err := RenderTableWithRandomUUIDTemplate(table, storeUUID, tplCopyInto)
+	require.NoError(t, err)
+	mergeInto, err := RenderTableWithRandomUUIDTemplate(table, storeUUID, tplMergeInto)
+	require.NoError(t, err)
 
 	// Note the intentional missing semicolon, as this is a subquery.
 	require.Equal(t, `
-		SELECT 123, test_table.flow_document
-		FROM test_table
-		JOIN (
-			SELECT $1[0] AS key1, $1[1] AS key2
-			FROM @flow_v1/00010203-0405-0607-0809-0a0b0c0d0e0f
-		) AS r
-		ON test_table.key1 = r.key1 AND test_table.key2 = r.key2
-		`,
-		keyJoinSQL)
+	SELECT 0, test_table.flow_document
```

Comment:
It might be nice to have these outputs in a snapshot like we have done for bigquery and postgres. It is nice to see the diff side-by-side though, so I'm fine either way.